docs/src/routes/documentation/NodeGroup/Example5.js

```diff
@@ -0,0 +1,171 @@
+// @flow weak
+/* eslint react/no-multi-comp: "off", max-len: "off" */
+
+import { interpolate } from 'd3-interpolate';
+import { scaleOrdinal } from 'd3-scale';
+import { arc, pie } from 'd3-shape';
+import Surface from 'docs/src/components/Surface';
+import React, { PureComponent } from 'react';
+import NodeGroup from 'resonance/NodeGroup';
+
+const colors = scaleOrdinal()
+  .range([
+    '#6C6B74',
+    '#2E303E',
+    '#9199BE',
+    '#54678F',
+    '#212624',
+    '#3A4751',
+    '#1E272E',
+    '#0A151D',
+    '#030C12',
+    '#253517',
+    '#5D704E',
+    '#324027',
+    '#19280C',
+    '#0D1903',
+  ]);
+
+// **************************************************
+//  SVG Layout
+// **************************************************
+const view = [1000, 250];      // [width, height]
+const trbl = [10, 10, 10, 10]; // [top, right, bottom, left] margins
+
+const dims = [ // Adjusted dimensions [width, height]
+  view[0] - trbl[1] - trbl[3],
+  view[1] - trbl[0] - trbl[2],
+];
+
+// **************************************************
+//  Mock Data
+//  Data is being generated by Math.random in the
+//  constructor and randomized in update
+// **************************************************
+const mockData = [
+  {
+    name: 'Linktype',
+  }, {
+    name: 'Quaxo',
+  }, {
+    name: 'Skynoodle',
+  }, {
+    name: 'Realmix',
+  }, {
+    name: 'Jetpulse',
+  }, {
+    name: 'Chatterbridge',
+  }, {
+    name: 'Riffpedia',
+  }, {
+    name: 'Layo',
+  }, {
+    name: 'Oyoba',
+  }, {
+    name: 'Ntags',
+  },
+];
+
+const getArc = pie()
+  .value((d) => d.value)
+  .sort(null);
+
+const arcPath = arc()
+  .innerRadius((dims[1] / 3.5))
+  .outerRadius((dims[1] / 2));
+
+class Example extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const data = mockData.map((n) => {
+      return { ...n, value: Math.random() };
+    });
+
+    const newArcs = getArc(data);
+    this.state = {
+      data,
+      arcs: newArcs,
+      arcsCache: newArcs.reduce((m, n) => {
+        return Object.assign(m, { [n.data.name]: n });
+      }, {}),
+    };
+  }
+
+  update = () => {
+    this.setState((prevState) => {
+      const data = prevState.data.map((n) => {
+        return { ...n, value: Math.random() };
+      });
+
+      return {
+        data,
+        arcs: getArc(data),
+      };
+    });
+  };
+
+  arcTween = (beg, end, name) => {
+    const i = interpolate(beg, end);
+
+    return (t) => {
+      this.setState((prevState) => {
```

Comment:
Man,  you are totally right!  I wasn't even looking at that.  I just stepped back and thought about it.  There's a completely different (and much easier) way to do this.  No arcs cache even.

Try this code out when you have a chance.  It think it handles your scenario and I also made it handle cases where arcs are entering and exiting (kind of depends on how you want that to work - didn't really spend much time on that).  I also added a little ease function on there 👍 

```
import { scaleOrdinal } from 'd3-scale';
import { easeExpInOut } from 'd3-ease';
import { arc, pie } from 'd3-shape';

import Surface from 'docs/src/components/Surface';
import React, { PureComponent } from 'react';
import NodeGroup from 'resonance/NodeGroup';

const colors = scaleOrdinal()
  .range([
    '#a6cee3',
    '#1f78b4',
    '#b2df8a',
    '#33a02c',
    '#fb9a99',
    '#e31a1c',
    '#fdbf6f',
    '#ff7f00',
    '#cab2d6',
    '#6a3d9a',
  ]);

// **************************************************
//  SVG Layout
// **************************************************
const view = [1000, 250];      // [width, height]
const trbl = [10, 10, 10, 10]; // [top, right, bottom, left] margins

const dims = [ // Adjusted dimensions [width, height]
  view[0] - trbl[1] - trbl[3],
  view[1] - trbl[0] - trbl[2],
];

// **************************************************
//  Mock Data
//  Data is being generated by Math.random in the
//  constructor and randomized in update
// **************************************************
const mockData = [
  {
    name: 'Linktype',
  }, {
    name: 'Quaxo',
  }, {
    name: 'Skynoodle',
  }, {
    name: 'Realmix',
  }, {
    name: 'Jetpulse',
  }, {
    name: 'Chatterbridge',
  }, {
    name: 'Riffpedia',
  }, {
    name: 'Layo',
  }, {
    name: 'Oyoba',
  }, {
    name: 'Ntags',
  },
];

const pieLayout = pie()
  .value((d) => d.value)
  .sort(null);

const arcPath = arc()
  .innerRadius((dims[1] / 3.5))
  .outerRadius((dims[1] / 2));

class Example extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      arcs: this.getArcs(),
    };
  }

  getArcs = () => {
    const data = mockData.map(({ name }) => ({ name, value: Math.random() }));
    return pieLayout(data);
  }

  update = () => {
    this.setState(() => ({
      arcs: this.getArcs(),
    }));
  }

  render() {
    const { arcs } = this.state;

    return (
      <div>
        <button onClick={this.update}>
          Randomize
        </button>
        <Surface view={view} trbl={trbl}>
          <g transform={`translate(${dims[0] / 2}, ${dims[1] / 2})`}>
            <NodeGroup
              data={arcs}
              keyAccessor={(d) => d.data.name}

              start={({ startAngle }) => ({
                startAngle,
                endAngle: startAngle,
              })}

              enter={({ endAngle }) => ({
                endAngle: [endAngle],
                timing: { duration: 1500, ease: easeExpInOut },
              })}

              update={({ startAngle, endAngle }) => ({
                startAngle: [startAngle],
                endAngle: [endAngle],
                timing: { duration: 1500, ease: easeExpInOut },
              })}

              leave={({ endAngle }, index, remove) => ({
                startAngle: [endAngle],
                endAngle: [endAngle],
                timing: { duration: 1500, ease: easeExpInOut },
                events: { end: remove },
              })}

              render={({ data: { name } }, state) => {
                const { startAngle, endAngle } = state;

                return (
                  <path
                    fill={colors(name)}
                    d={arcPath({ startAngle, endAngle })}
                  />
                );
              }}
            />
          </g>
        </Surface>
      </div>
    );
  }
}

export default Example;
```